contrib/archivetest.c

```diff
@@ -201,7 +201,6 @@ int main(int argc, char *argv[])
 			v_print(v, "skipping");
 		} else {
 			while ((r = archive_read_data(a, buffer, 4096) > 0))
```

Comment:
It looks like this expression is actually wrong.  I think it should be
```
   (r = archive_read_data(a, buffer, 4096)) > 0
```
to ensure the assignment happens before the comparison.